pkg/blockstorage/azure/auth.go

```diff
@@ -25,6 +28,28 @@ func isMSICredsAvailable(config map[string]string) bool {
 		config[blockstorage.AzureClientSecret] == "")
 }
 
+type ClientCredentialsConfig struct {
+	ClientID     string
+	ClientSecret string
+	TenantID     string
+	AuxTenants   []string
+	AADEndpoint  string
+	Resource     string
+}
+
+// Defaults to Public Cloud and Resource Manager Endpoint.
+func NewClientCredentialsConfig(clientID string, clientSecret string, tenantID string) ClientCredentialsConfig {
```

Comment:
Yes correct, As of now I didn't find any direct replacement of this struct.
So instead of using this struct, we refer to the attributes by directly using the keys from creds map and use them wherever needed, e.g. `Authenticate` method

Reply:
> So instead of using this struct, we can refer to the attributes by directly using the keys from creds map and use them wherever needed, e.g. Authenticate method.
Do let me know if this sounds okay.

I think it's a good idea! Though might be a big scope creep for this PR. I did a quick look through the call stack and it might be a bit of refactoring to accomplish that. I suggest we take note of this and address in a different PR